USSF-ORBIT/ussf-portal-cms
# chore: Upgrade keystone and node

# SC-244

## Proposed changes

* Upgrades to Node 18
* Upgrades Keystone to latest major release with breaking changes

## Reviewer notes

There have been updates made to resolve breaking changes related to:

* Lists - types, and access control of users
* **All** unit tests -- test env setup, updated error messages
* Redis sessions
* GraphQL schema extension
* tsconfig.json settings 

## Setup


* In ussf-portal-cms, run `yarn services:up && yarn dev`
* In ussf-portal-client, run `yarn dev`

## Code review steps
* Log in as different users/roles and ensure access is working as expected
* Create and edit different list objects, ensure functionality is working as expected
* Make sure tests pass
* Make sure portal works as expected

### As the original developer, I have

  - ... 
  - [ ] Including [axe-playwright](https://github.com/abhinaba-ghosh/axe-playwright) checks when UI changes
- Performed [a11y testing](https://github.com/trussworks/accessibility/blob/master/sample_a11y_testing_process.md):
- For any new migrations/schema changes:

### As code reviewer(s), I have

- Performed [a11y testing](https://github.com/trussworks/accessibility/blob/master/sample_a11y_testing_process.md):

### As a designer reviewer, I have

- Performed [a11y testing](https://github.com/trussworks/accessibility/blob/master/sample_a11y_testing_process.md):

---

## Screenshots
